app/code/Magento/ConfigurableProduct/Model/ResourceModel/Product/Type/Configurable/Product/Collection.php

```diff
@@ -79,12 +79,12 @@ protected function _renderFilters()
         parent::_renderFilters();
         $metadata = $this->getProductEntityMetadata();
         $parentIds = [];
-        foreach ($this->products as $product) {
-            $parentIds[] = $product->getData($metadata->getLinkField());
+        if ($this->products) {
+            foreach ($this->products as $product) {
+                $parentIds[] = $product->getData($metadata->getLinkField());
+            }
         }
-
         $this->getSelect()->where('link_table.parent_id in (?)', $parentIds);
```

Comment:
What if `$parentIds` is an empty array?

Reply:
not sure what select would do, but yes, it would totally make sense to move it into the if because it is empty anyway when the if is skipped